src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -372,14 +402,27 @@ public boolean isRequestCacheable(
     public void purgeEntriesForNonPendingRequests(
         Set<Integer> requestHashes)
     {
-        cachedEntriesByRequestHash.forEach((requestHash, cacheEntry) ->
+        for (final MutableInteger count = new MutableInteger(1); count.value <= allowedCacheEvictionCount;)
         {
-            if (!requestHashes.contains(requestHash))
+            final FrequencyBucket frequencyBucket = frequencies.get(count.value);
+            frequencyBucket.entries().forEach(entry ->
```

Comment:
Perhaps just use an explicit iterator with a nested loop instead of `forEach` with lambda, then change `MutableInteger` to regular `int`.